source/MRMesh/MRMeshDecimateCallbacks.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "MRMesh/MRMesh.h"
+
+namespace MR
+{
+// there is a vertex parameter that needs to be recalculated when collapsing the edge
+template <typename T>
+static auto preCollapseVertAttribute( const Mesh& mesh, Vector<T, VertId>& data )
```

Comment:
no need in `static` for templates.
The names `preCollapseVertAttribute` and `meshAttributesUpdatePreCollapseCb` are too distinct. Please make them closer.